sources/packages/backend/apps/api/src/services/application-offering-change-request/application-offering-change-request.service.ts

```diff
@@ -515,20 +523,131 @@ export class ApplicationOfferingChangeRequestService {
    * @param applicationOfferingChangeRequestId application offering change request id for which to update the status.
    * @param applicationOfferingChangeRequestStatus the application offering change request status to be updated.
    * @param studentConsent student consent to approve the application offering change request.
+   * @param auditUserId user that should be considered the one that is causing the changes.
    */
   async updateApplicationOfferingChangeRequest(
     applicationOfferingChangeRequestId: number,
     applicationOfferingChangeRequestStatus: ApplicationOfferingChangeRequestStatus,
     studentConsent: boolean,
+    auditUserId: number,
   ): Promise<void> {
+    const auditUser = { id: auditUserId } as User;
+    const currentDate = new Date();
     await this.applicationOfferingChangeRequestRepo.update(
       {
         id: applicationOfferingChangeRequestId,
       },
       {
         applicationOfferingChangeRequestStatus,
         studentConsent,
+        studentActionDate: currentDate,
+        modifier: auditUser,
+        updatedAt: currentDate,
       },
     );
   }
+
+  /**
+   * Assess the application offering change request status for the given application offering change request id for the ministry user.
+   * @param applicationOfferingChangeRequestId application offering change request id for which to update the status.
+   * @param applicationOfferingChangeRequestStatus the application offering change request status to be updated.
+   * @param assessmentNote note added while updating the application offering change request.
+   * @param userId user id of the user making the approve or decline request.
+   */
+  async assessApplicationOfferingChangeRequest(
+    applicationOfferingChangeRequestId: number,
+    applicationOfferingChangeRequestStatus:
+      | ApplicationOfferingChangeRequestStatus.Approved
+      | ApplicationOfferingChangeRequestStatus.DeclinedBySABC,
+    assessmentNote: string,
+    userId: number,
+  ): Promise<void> {
+    const auditUser = { id: userId } as User;
+    const currentDate = new Date();
+    const application = new Application();
+    const applicationOFferingChangeRequest =
+      await this.applicationOfferingChangeRequestRepo.findOne({
+        select: {
+          id: true,
+          requestedOffering: {
+            id: true,
+          },
+          application: {
+            id: true,
+            student: {
+              id: true,
+              user: {
+                id: true,
+                firstName: true,
+                lastName: true,
+              },
+            },
+          },
+        },
+        relations: {
+          application: {
+            student: {
+              user: true,
+            },
+          },
+          requestedOffering: true,
+        },
+        where: {
+          id: applicationOfferingChangeRequestId,
+        },
+      });
```

Comment:
We can use the method `getApplicationOfferingChangeRequestStatusById` and avoid this.

Reply:
This one only returns the status only. Did you mean `getById` ?

Reply:
Oops! yes I meant getById

Reply:
Got it. Thanks @dheepak-aot